libraries/chain/combined_database.cpp

```diff
@@ -183,33 +196,30 @@ namespace eosio { namespace chain {
       case backing_store_type::ROCKSDB:
          try {
             try {
-               kv_undo_stack->write_state();
-               kv_database->flush(true, true);
+               kv_database->flush();
             }
             FC_LOG_AND_RETHROW()
          }
          CATCH_AND_EXIT_DB_FAILURE()
       }
    }
 
-   std::unique_ptr<kv_context> combined_database::create_kv_context(name receiver,
-                                                                    kv_resource_manager resource_manager,
+   std::unique_ptr<kv_context> combined_database::create_kv_context(name receiver, kv_resource_manager resource_manager,
                                                                     const kv_database_config& limits) {
       switch (backing_store) {
          case backing_store_type::ROCKSDB:
-            return create_kv_rocksdb_context<b1::chain_kv::view, b1::chain_kv::write_session, kv_resource_manager>(
-                  *kv_database, *kv_undo_stack, receiver, resource_manager, limits);
+            return create_kv_rocksdb_context<session_type, kv_resource_manager>(kv_undo_stack->top(), receiver,
```

Comment:
@larryk85 @linhuang-blockone @revl 
I'm running into an exception where there is no top of the undo stack at this point.  I'm thinking that I might need to push a new session but I'm not quite sure where that should be done.  I was thinking one of two places:

1.  I can push a session right before this line.  
2.  I can push a new session when the combined database instance is creating.

With both of these places, would I always be pushing a session or would I only push a session when the undo stack is empty?